libparsec/crates/types/src/local_device.rs

```diff
@@ -60,13 +62,29 @@ impl LocalDevice {
             signing_key: signing_key.unwrap_or_else(SigningKey::generate),
             private_key: private_key.unwrap_or_else(PrivateKey::generate),
             initial_profile,
-            user_realm_id: VlobID::default(),
-            user_realm_key: SecretKey::generate(),
+            user_realm_id: user_realm_id.unwrap_or_default(),
+            user_realm_key: user_real_key.unwrap_or_else(SecretKey::generate),
             local_symkey: SecretKey::generate(),
             time_provider: time_provider.unwrap_or_default(),
         }
     }
 
+    /// generates a new device for a given user
+    pub fn from_existing_device_for_user(
```

Comment:
Is the suffix `_for_user` useful ? for me a device is always linked to an user/machine

Reply:
It's to convey that it's for the *same* user

Reply:
But that's already implied by `existing_device`, no ? :thinking:

Reply:
I don't know, it feels more explicit that way IMHO